rllib/policy/torch_policy.py

```diff
@@ -133,33 +129,60 @@ def __init__(
         #   parallelization will be done.
         # TODO: (sven) implement data pre-loading and n loader buffers for
         #  torch.
-        if config["_fake_gpus"] or config["num_gpus"] == 0 or \
-                not torch.cuda.is_available():
+
+        # Get devices to build the graph on.
+        worker_idx = self.config.get("worker_index", 0)
+        if not config["_fake_gpus"] and \
+                ray.worker._mode() == ray.worker.LOCAL_MODE:
+            num_gpus = 0
+        elif worker_idx == 0:
+            num_gpus = config["num_gpus"]
+        else:
+            num_gpus = config["num_gpus_per_worker"]
+        gpu_ids = list(range(torch.cuda.device_count()))
```

Comment:
I don't think we can get all devices directly here.
Image that if we run the driver on the 5 devices node, and a remote worker is also scheduled to this node, all 5 devices are available to the remote worker, which I don't think make sense.

Reply:
The environment variable `CUDA_VISIBLE_DEVICES` is set for the remote worker. `torch.cuda.device_count()` will respect the `CUDA_VISIBLE_DEVICES` and return the number of CUDA visible GPUs.